src/main/kotlin/app/revanced/patcher/patch/BytecodePatchContext.kt

```diff
@@ -98,8 +98,23 @@ class BytecodePatchContext internal constructor(private val config: PatcherConfi
      * @param predicate A predicate to match the class.
      * @return A proxy for the first class that matches the predicate.
      */
-    fun classBy(predicate: (ClassDef) -> Boolean) =
-        classes.proxyPool.find { predicate(it.immutableClass) } ?: classes.find(predicate)?.let { proxy(it) }
+    fun classBy(predicate: (ClassDef) -> Boolean): ClassProxy? {
+        synchronized(classes.proxyPool) {
+            val proxy = classes.proxyPool.find { predicate(it.immutableClass) }
+            if (proxy != null) return proxy;
+        }
+
+        val classDef: ClassDef?
+        synchronized(classes) {
```

Comment:
Is synchronization needed here? Patches can also access the classes list, should they also synchronize?